File: ruby/JsonStrings.rb
def good_json
  <<-EOF
  {
    "name": "Nooby McNoobington",
    "winPercent": null
  }
  EOF
end

def glad_json
  <<-EOF
  {
    "name": "Mikhail Tal",
    "winPercent": 85.2
  }
  EOF
end

def sad_json
  <<-EOF
  {
    "name": 123,
    "winPercent": false
  }
  EOF
end

def bad_json
  <<-EOF
  {
    "name": "Nooby McNoobington",
    'winPercent': null
  }
  EOF
end

def rad_json
  <<-EOF
  {
    "fiz": 123,
    "faz": null,
    "guz": "Boom!"
  }
  EOF
end

File: swift/jsonstrings.swift
//multiline literals must start and end on their own lines
let goodJson = """
{
    "name": "Nooby McNoobington",
    "winPercent": null
}
"""

let gladJson = """
{
    "name": "Mikhail Tal",
    "winPercent": 85.2
}
"""

let deadJson = """
{
    "name": "Never McWonAGame",
    "winPercent": 0
}
"""

let sadJson = """
{
    "name": "Nooby McNooberson",
    "winPercent": false
}
"""

let badJson = """
{
    "name": "Nooby McNooberson",
    'winPercent': 13.3
}
"""

let radJson = """
{
    "biz": "Nooby McNooberson",
    "baz": 13.3
}
"""
